client/my-sites/current-site/index.jsx

```diff
@@ -159,9 +156,13 @@ const CurrentSite = React.createClass( {
 // TODO: make this pure when sites can be retrieved from the Redux state
 module.exports = connect(
 	( state, ownProps ) => {
+		const selectedSiteId = getSelectedSiteId( state );
 		const selectedSite = getSelectedSite( state );
+
 		return {
-			selectedSite: selectedSite || ownProps.sites.getPrimary()
+			selectedSiteId,
```

Comment:
Oh that make a lot more sense. I misread the intent in the render section: https://github.com/Automattic/wp-calypso/pull/10481/files#diff-41b8f70d120b8ee4a12b8d6b635ccb92L140